jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/DefinitionService.java

```diff
@@ -41,13 +42,14 @@
 	 * @param deploymentId identifier of deployment this process belongs to, 
 	 * 			might be null if built definition does not need to be stored
 	 * @param bpmn2Content actual BPMN xml content as string to be parsed and processed
-	 * @param classLoader class loader that should be used while parsing the BPMN2 in case custom classes are referenced
+	 * @param kieContainer the {@link KieContainer} instance that contains the deployment project: this should be used when 
+	 *          parsing the BPMN2 in case custom classes or other project resources (processes, rules) are referenced
 	 * @param cache indicates if the definition service should cache this <code>ProcessDefinition</code>
 	 * @return fully populated <code>ProcessDefinition</code>
 	 * @throws IllegalArgumentException in case build operation cannot be completed
 	 */
 	ProcessDefinition buildProcessDefinition(String deploymentId, String bpmn2Content,
-			ClassLoader classLoader, boolean cache) throws IllegalArgumentException;
+			KieContainer kieContainer, boolean cache) throws IllegalArgumentException;
```

Comment:
what is the reason to change classLoader to kieContainer? Asking mainly as this changes the api interface that can be used by users since it is public api for jbpm services - used for example by ejb. Most likely it's not so frequently used as standalone so most likely not big impact but the bigger issue might be that we expect users to always have kieContainer to use this method. While previously it was completely kieContainer agnostic.